File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/RolePermissionIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;

class RolePermissionIdFlattenedDto
{

    const PROPERTIES = [
            'roleId' => 'string',
            'permissionId' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var RolePermissionId
     */
    private $value;

    /**
     * @param RolePermissionId $value
     */
    public function __construct(RolePermissionId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new RolePermissionId();
        }
    }

    /**
     * @return RolePermissionId
     */
    public function toRolePermissionId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getRoleId()
    {
        return $this->value->getRoleId();
    }

    /**
     * @param string $roleId
     */
    public function setRoleId($roleId)
    {
        $this->value->setRoleId($roleId);
    }

    /**
     * @return string
     */
    public function getPermissionId()
    {
        return $this->value->getPermissionId();
    }

    /**
     * @param string $permissionId
     */
    public function setPermissionId($permissionId)
    {
        $this->value->setPermissionId($permissionId);
    }


}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/RolePermissionIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class RolePermissionIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "roleId",
            "permissionId",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private RolePermissionId value;

    public RolePermissionIdFlattenedDto()
    {
        this(new RolePermissionId());
    }

    public RolePermissionIdFlattenedDto(RolePermissionId value)
    {
        this.value = value;
    }

    public RolePermissionId toRolePermissionId()
    {
        return this.value;
    }


    public String getRoleId()
    {
        return this.value.getRoleId();
    }

    public void setRoleId(String roleId)
    {
        this.value.setRoleId(roleId);
    }

    public String getPermissionId()
    {
        return this.value.getPermissionId();
    }

    public void setPermissionId(String permissionId)
    {
        this.value.setPermissionId(permissionId);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        RolePermissionIdFlattenedDto other = (RolePermissionIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

